js.go

```diff
@@ -2105,9 +2105,13 @@ func checkMsg(msg *Msg, checkSts bool) (usrMsg bool, err error) {
 	case reqTimeoutSts:
 		// Older servers may send a 408 when a request in the server was expired
 		// and interest is still found, which will be the case for our
-		// implementation. Regardless, ignore 408 errors, the caller will
-		// go back to wait for the next message.
-		err = nil
+		// implementation. Regardless, ignore 408 errors until receiving at least
```

Comment:
Alternatively, set err to ErrTimeout here, (without the need of cancelWhen408), but just treat the possible ErrTimeout from the caller perspective, see below.